presto-main/src/main/java/com/facebook/presto/execution/CreateViewTask.java

```diff
@@ -88,9 +90,14 @@ public String explain(CreateView statement, List<Expression> parameters)
                 .map(field -> new ViewColumn(field.getName().get(), field.getType()))
                 .collect(toImmutableList());
 
+        List<ColumnMetadata> columnMetadata = columns.stream()
+                .map(column -> new ColumnMetadata(column.getName(), column.getType()))
+                .collect(toImmutableList());
+
         String data = codec.toJson(new ViewDefinition(sql, session.getCatalog(), session.getSchema(), columns, Optional.of(session.getUser())));
+        ConnectorTableMetadata viewMetadata = new ConnectorTableMetadata(name.asSchemaTableName(), columnMetadata);
```

Comment:
nit: move this above where data is set, so this is right next to the columnMetadata creation and the createView() arguments are created in the order they appear.